### PR TITLE
EduPlanner 📆 BackEnd application V.1.0.0 🧱 

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR We present the Edu Planner Appointments Application:🌐 
**This BackEnd Part of the application has undergone its development 👨‍💻 phase and completed successfully**

## Project  KanBan Board
board

## Requirements
- All requirements can be found [here](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)

## ACHIEVEMENTS ⛰️
- Application has a fully built and well-designed UI🎨 and UX⚙️ system ready to fulfill all these [requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)
- Link To Application FrontEnd Deployed app [here](https://edu-planner.onrender.com)
- Link To Application BackEnd Deployed app [here](https://edu-planner-backend.onrender.com/)
- Link to Application FrontEnd SourceCode [here](https://github.com/ab-noori/EduPlanner-Frontend)

## NOTE
- In this Project We worked As a Team Of three (@gisachris , @ab-noori , @BANCUNGUYE66 ) and Implemented the Project Requirements to the Best of our Abilities and According to the Project Requirements for a `Groups of three`.

## Conclusion
**We would love to hear 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the site thank you! And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+We welcome any improvements if your changes are good and align with the project's goals.
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (Meltrust) [inline README.md]: - [ ] This link to the LICENSE file does not work.  Kindly, link to the correct file to correct it. 👍 

image

File: README.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+Original design idea by [ Murat Korkmaz on Behance.](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## FAQ <a name="faq"></a>
+
+- **Can I use this repository?**
+
+Please feel free to fork this repository for your own projects.
```
Comment (Meltrust) [inline README.md:156]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

File: app/controllers/api/reservations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      render json: @reservation
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /reservations/1
+  def destroy
+    @reservation.destroy
+  end
```
Comment (Meltrust) [inline app/controllers/api/reservations_controller.rb]: - [ ] What happens is the reservation is not successfully destroyed?  Kindly, add some error handling. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @gisachris ,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by ab-noori** [comment]
**Hi @Meltrust 👋,**
Thank you so much 🙏 for reviewing our pull request.
Your nice feedback is really appreciated 🙏.
Our team will work on the requested changes 🎯

thank-you-gif-22

**Comment by gisachris** [comment]
## IN This PR We present the Edu Planner Appointments Application Required Changes(Backend): ♻️
**This application has undergone Making changes ♻️  and has completed Successfully**

## Changes Required
- Implement Error handling on the Backend of the Application.
- Remove `MiniTest` files since we are using `Rspec` for Testing. 🧪 
- Put Proper links to `Licence` file 📜

## ACHIEVEMENTS ⛰️
- All the Above Where achieved. ✅ 

## Conclusion
**We would love to hear 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the site thank you! And we look forward to hearing from you soon.**  🙇

**Review by Meltrust — APPROVED** [review]
#### Hi @gisachris,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause





Cheers and Happy coding!👏👏👏


------

**Comment by gisachris** [comment]
@Meltrust  Thank you for the kind review 🤝

**Pull request merged**